data/core/process.lua

```diff
@@ -0,0 +1,87 @@
+local config = require "core.config"
+
+process.stream = {}
+process.stream.__index = proc.stream
+
+function process.stream.new(proc, fd)
+  return setmetatable({ fd = fd, process = proc, buf = {}, len = 0 }, proc.stream)
+end
+
+-- read function can take either "*all", "*line", or a byte count.
+function process.stream:read(bytes, options)
```

Comment:
Please write proper [luacats](https://luals.github.io/wiki/annotations/) (hell yeah) annotations, so that we don't have to revisit some time later and write it.
Don't forget to document `options` as well.

Reply:
Ugh.